android/KMEA/app/src/main/assets/android-host.js

```diff
@@ -25,27 +25,28 @@ function init() {
   //document.body.style.backgroundColor="transparent";
   //window.console.log('Device type = '+device);
   //window.console.log('Keyboard height = '+oskHeight);
-  keyman.init({'app':device,'fonts':'packages/',root:'./'});
-  keyman.util.setOption('attachType','manual');
-  keyman.oninserttext = insertText;
-  window.showKeyboardList = showMenu;
-  window.hideKeyboard = hideKeyboard;
-  window.menuKeyUp = menuKeyUp;
+  keyman.showKeyboardList = showMenu;
+  keyman.hideKeyboard = hideKeyboard;
+  keyman.menuKeyUp = menuKeyUp;
   keyman.getOskHeight = getOskHeight;
   keyman.getOskWidth = getOskWidth;
   keyman.beepKeyboard = beepKeyboard;
-  var ta = document.getElementById('ta');
-  keyman.setActiveElement(ta);
-
-  ta.readOnly = false;
-  checkTextArea();
-
-  // Tell KMW the default banner height to use
-  com.keyman.osk.Banner.DEFAULT_HEIGHT =
-    Math.ceil(window.jsInterface.getDefaultBannerHeight() / window.devicePixelRatio);
+  keyman.init({
+    'embeddingApp':device,
+    'fonts':'packages/',
+    oninserttext: insertText,
+    root:'./'
+  }).then(() => {
+    const bannerHeight = Math.ceil(window.jsInterface.getDefaultBannerHeight() / window.devicePixelRatio);
+
+    // The OSK is not available until initialization is complete.
+    console.log(keyman.osk.bannerView);
+    keyman.osk.bannerView.activeBannerHeight = bannerHeight;
+    keyman.osk.refreshLayout();
```

Comment:
Why is `refreshLayout()` needed here for the Android Engine but not in the same place for iOS engine?